lib/importer/factory/object_factory.rb

```diff
@@ -124,7 +124,9 @@ def file_spec(file_path)
           # Hyrax encodes the URL here https://github.com/samvera/hyrax/blob/4fd8d9ad3c32db7deffc3b5246af5d1459a4b046/app/actors/hyrax/actors/create_with_remote_files_actor.rb#L53
           # but the presigned url we get above is already encoded. So we're unencoding it here before we sent it to
           # hyrax, where it will get encoded again. Not pretty, but it works
-          url = Addressable::URI.unencode(url)
+          # 2018-1-22 Update: sort of a weird situation, but commenting out the 'unencode' below allows the URL to resolve on AWS, but not locally
+          # I'm going to leave it commented out for now, but this is an annoying environment mismatch we'll have to address later on
+          url = Addressable::URI.unencode(url) if Rails.env.development?
```

Comment:
If commenting out the line makes this work in AWS, than I think we want `unless Rails.env.production?` because we need this set in `development` and `test`, and staging and production are both using the `production` environment.

Reply:
right now our AWS environment is set to staging and we might have a staging area in AWS in the future so we'd want to check for at least those two. And for testing, are we actually grabbing the file from minio/s3? Ideally all that would be mocked because you don't want to be making real requests in the test environment.

although i think i just remembered we *are* actually hitting minio in the test environment, so i'll update the check for dev and test!

Reply:
Just checked this on AWS staging:
```
Rails.env
=> "production"
```

Reply:
yeah you're right, i was thinking of this line: https://github.com/nulib/donut/blob/master/config/environments/production.rb#L2

we have that set to true in AWS right now, but the actual rails env in AWS is production. I'll change line to `unless Rails.env.production?`